back/engines/free/email_campaigns/app/models/concerns/email_campaigns/activity_triggerable.rb

```diff
@@ -6,10 +6,37 @@ module ActivityTriggerable
 
     included do
       before_send :filter_activity_triggered
+      before_send :filter_activity_too_old
     end
 
     def filter_activity_triggered(activity:, time: nil)
       activity && activity_triggers.dig(activity.item_type, activity.action)
     end
+
+    # Skip activities that are older than a reasonable threshold
+    # to prevent reprocessing of old activities
+    def filter_activity_too_old(activity:, time: nil)
+      return false if activity.nil?
+
+      # Only report if the activity is actually too old
+      # Report so that we can examine the backtraces to get more context
+      # on why this is happening (see: TAN-4469)
+      if activity.acted_at < 7.days.ago
```

Comment:
I was a bit confused at first because in my mind this read like "acted less than 7 days ago", but it's actually more something like "acted before 7 days ago".

Reply:
Hopefully, the related test documents this well enough.